__tests__/integration/mint-tokens.test.js

```diff
@@ -255,25 +266,87 @@ describe('mint token', () => {
     expect(p2pkh.address.base58).toEqual(address0);
   });
 
+  it('should mint and create data outputs', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/mint-tokens')
+      .send({
+        token: tokenA.uid,
+        amount: 100,
+        data: ['foobar1', 'foobar2'],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    // Current funds: 9 HTR + 960 TKA
+
+    const transaction = response.body;
+    expect(transaction.success).toBe(true);
+    // If unshift_data is not specified, the data output will be the first output
+    const dataOutput1 = transaction.outputs[1];
+    const dataOutput2 = transaction.outputs[0];
+
+    await TestUtils.waitForTxReceived(wallet1.walletId, response.body.hash);
+
+    console.log(`Data output foobar1: ${JSON.stringify(dataOutput1)}`);
+    console.log(`Data output foobar2: ${JSON.stringify(dataOutput2)}`);
+    const script1 = Array.from((new ScriptData('foobar1')).createScript());
+    const script2 = Array.from((new ScriptData('foobar2')).createScript());
+
+    expect(dataOutput1.token_data).toBe(0);
+    expect(dataOutput1.value).toBe(1);
+    expect(dataOutput1.script.data).toEqual(script1);
+
+    expect(dataOutput2.token_data).toBe(0);
+    expect(dataOutput2.value).toBe(1);
+    expect(dataOutput2.script.data).toEqual(script2);
+  });
+
+  it('should mint and create a data output at first position', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/mint-tokens')
+      .send({
+        token: tokenA.uid,
+        amount: 100,
+        data: ['foobar'],
+        unshift_data: false,
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    // Current funds: 9 HTR + 960 TKA
+
+    const transaction = response.body;
+    expect(transaction.success).toBe(true);
+    const dataOutput = transaction.outputs[transaction.outputs.length - 1];
+    const script = Array.from((new ScriptData('foobar')).createScript());
+
+    await TestUtils.waitForTxReceived(wallet1.walletId, response.body.hash);
+
+    console.log(JSON.stringify(dataOutput));
+    expect(dataOutput.token_data).toBe(0);
+    expect(dataOutput.value).toBe(1);
+    expect(dataOutput.script.data).toEqual(script);
```

Comment:
Same comment as above regarding the balance

Reply:
Done